sconfig/shared/src/test/scala/org/ekrich/config/impl/FormattingOptionsTest.scala

```diff
@@ -0,0 +1,108 @@
+package org.ekrich.config.impl
+import org.junit.Assert.*
+import org.junit.*
+import org.ekrich.config.{
+  ConfigFactory,
+  ConfigParseOptions,
+  ConfigRenderOptions,
+  FormattingOptions
+}
+
+class FormattingOptionsTest extends TestUtilsShared {
+  val parseOptions = ConfigParseOptions.defaults.setAllowMissing(true)
+  val myDefaultRenderOptions = ConfigRenderOptions.defaults
+    .setJson(false)
+    .setOriginComments(false)
+    .setComments(true)
+    .setFormatted(true)
+
+  def formatHocon(
+      str: String
+  )(implicit formattingOptions: FormattingOptions): String =
+    ConfigFactory
+      .parseString(str, parseOptions)
+      .root
+      .render(myDefaultRenderOptions.setFormattingOptions(formattingOptions))
+
+  @Test
+  def noNewLineAtTheEnd(): Unit = {
+    implicit val formattingOptions = FormattingOptions(newLineAtEnd = false)
+    val in = """r {
+               |}""".stripMargin
+    val result = formatHocon(in)
+    val expected = "r {}"
+    checkEqualObjects(result, expected)
+  }
+
+  @Test
+  def keepOriginOrderOfEntries(): Unit = {
```

Comment:
@ekrich check this one testcases.
It was the main reason why I've added new options.
Before always, entries were sorted in alphabetical order.